hazelcast-jet-core/src/main/java/com/hazelcast/jet/Sinks.java

```diff
@@ -58,6 +64,204 @@ private Sinks() {
         return new SinkImpl<>(sinkName, metaSupplier);
     }
 
+    /**
+     * Returns a sink that updates the items it receives into a Hazelcast
+     * {@code IMap} with the specified name and merge function. The function
+     * will be wrapped inside an {@link EntryProcessor}. The value extracted
```

Comment:
I'm not sure if we need to mention EntryProcessor here, it's an implementation detail. It should be mentioned that updates happen in partition thread and will not be concurrent with other updates to the same key.

It's also not clear what happens if the function returns null. Is the entry removed? This is a good example where they document what happens as code: https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#merge-K-V-java.util.function.BiFunction-